clang/lib/3C/ConstraintBuilder.cpp

```diff
@@ -285,6 +289,12 @@ class FunctionVisitor : public RecursiveASTVisitor<FunctionVisitor> {
                                             "accepting var args.",
                                             E);
               } else {
+                if (PrintfStringArgIndices.find(I) !=
+                    PrintfStringArgIndices.end()) {
+                  // In `printf("%s", foo)`, foo should be an _Nt_array_ptr
```

Comment:
I might change this comment slightly to emphasize that each argument is treated individually. E.g., writing
```
// In `printf("... %s ...", ..., foo, ...)`, foo should be an _Nt_array_ptr
```
I was a little confused about this when I first looked at the code, so this was a concrete issue for me. :)

Reply:
Done.